TFM 5.0.1 1.12 Release

Wild1145 opened the pull request:
So far the testing has been positive and no issues have been identified. One issue I've noticed in the PR that we may need to address as a wider issue is around the ReadME as the TravisCI link does not match as its branch specific. 

I think I've also potentially confused the version increments, this was supposed to be a 0.0.1 increment on the current release but I think I've mixed 5.0 and 5.1 up in a few places now...

StevenLawson committed:
Set up Travis CI for TotalFreedom organization.

Wild1145 committed:
Merged Devel into 1.12 Alpha Branch

Wild1145 committed:
Resolved issue whereby the editor didnt save when it should have

Wild1145 committed:
Updated default config to account for removal of MC Version field.

Commit:
Add files via upload

Commit:
Wrong repo

Wild1145 committed:
Updated OV command alias

Resolves #1652 from an earlier oversite on the re-naming of the main command.

Wild1145 committed:
Move to using Java 1.8 instead of 1.7 resolving #2148 and resolves #2150

Wild1145 committed:
Removed achievement on the cake command as there is no similar advancement that seems to fit. Resolves #2121

Wild1145 committed:
Added console to the default host sender list. Resolves #2123

Wild1145 committed:
[BLEEDING] Fixes to the POM and TotalFreedom command to fix the build properties not working properly.

Hockeyfan360 committed:
Improved JSON API. Resolves #1884. (#1991)

* Allows external apps to see online admins through API.

This commit allows external apps/plugins to see which admins are online through the HTTP API. This could be used to develop plugins quicker, without needing to integrate them directly into TFM.

* Fixed formatting error.

Wild1145 committed:
Fixes to the build info as suggested on the issue. Resolves #2119

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=2163) <br/>All committers have signed the CLA.

Wild1145 commented:
@OxLemonxO Are you able to sign the CLA please? IT appears that when you accidentally pushed to this branch it kinda broke things...

JeromSar commented on pom.xml:
```diff
@@ -210,26 +237,66 @@
+                    <failOnNoGitDirectory>true</failOnNoGitDirectory>
+                    <failOnUnableToExtractRepoInfo>false</failOnUnableToExtractRepoInfo>
+                    <excludeProperties>
+                        <excludeProperty>git.branch</excludeProperty>
```
Considering we also have an `includeOnlyProperties`, is this list not redundant. It seems to me like there must be a better way to only include the right properties without excluding every other property explicitly.

Wild1145 replied:
I've just done some testing and it looks like that is the case, it adds a bit more info to the git.properties file but nothing too major.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java:
```diff
@@ -259,22 +259,31 @@ public void load(TotalFreedomMod plugin)
                     props = new Properties();
                     props.load(in);
                 }
+                try (InputStream in = plugin.getResource("git.properties"))
```
Is there any way where we can include the git properties in the build properties? From a logical perspective, git properties signify the build process/state and it would make sense to centralize this data in one file.

Wild1145 replied:
I did try to do this, but there was no easy way I could find (After 2 days of research) to do this without either not updating the git properties, or wiping the other build properties and wiping them back to their defaults. I'd like to have a better solution as this is certainly not perfect, but I couldn't come up with one unfortunately.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_cake.java:
```diff
@@ -43,12 +42,11 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
         return true;
     }
-}
+}
```
Should have a newline.

JeromSar commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_ov.java:
```diff
@@ -15,7 +15,7 @@
 
 @CommandPermissions(level = Rank.IMPOSTOR, source = SourceType.ONLY_IN_GAME)
-@CommandParameters(description = "Overlord - control this server in-game", usage = "access", aliases = "ov")
+@CommandParameters(description = "Overlord - control this server in-game", usage = "access", aliases = "overlord")
```
Perhaps these should be the other way around. The full filename should be `Command_overlord.java`, and the alias `ov`.

Wild1145 replied:
So this was just to fix a bug where the alias and file name were exactly the same. If you think we should change it I am inclined to say we should raise a new issue for the change and put it in a later release.

JeromSar replied:
If you fix something that's broken, fix it good then. From a gameplay perspective, it doesn't change much, solely the help pages.

Wild1145 replied:
The ticket requirements clearly state that the alias should be changed to /overlord from /ov as per the issue https://github.com/TotalFreedom/TotalFreedomMod/issues/1652 - That is what I have done.

JeromSar commented on src/main/resources/config.yml:
```diff
@@ -11,15 +11,15 @@ server:
   # All players who show up as owner
   owners:
-    - markbyron
+    - TheMinecraft
```
Unless you know who 'TheMinecraft' is, we might as well leave this an empty array.

ncurran02 replied:
TheMinecraft is Windows aka the owner of TF

Wild1145 replied:
Exactly what @Focusvity has said.

JeromSar replied:
okay!

JeromSar commented on pom.xml:
```diff
@@ -4,18 +4,19 @@
     <groupId>me.totalfreedom</groupId>
     <artifactId>totalfreedom</artifactId>
-    <version>5.0</version>
+    <version>5.1</version>
```
Considering this is still TFM Electrum, and not TFM _F_. We should stay at `5.0`.

Wild1145 replied:
Yes, I think I got some of the versioning confused over the last few months. Shall fix this.

JeromSar commented on pom.xml:
```diff
@@ -140,23 +137,51 @@
+                                <propertyfile file="src/main/resources/build.properties" comment="Build information. Edit this to your liking.">
                                     <entry key="buildAuthor" default="unknown" />
+                                    <entry key="buildNumber" default="0" />
+                                    <entry key="buildCodeName" default="Electrum (MC 1.12) Pre-Release 1" />
```
Is there any way not to hardcode the code name in the pom file? Could we instead hard code it in build.properties?

Wild1145 replied:
These are the build defaults for the POM. You can change them when the build.properties are generated but I wanted to ensure there are always default values as part of the generation.

JeromSar reviewed (changes requested):
Overall, good job!

Wild1145 committed:
Fixes to the versioning and some minor improvements in line with suggestions raised on the PR.

JeromSar commented:
Will look at this ASAP.

Telesphoreo commented:
I figured out how to merge the git properties into the build properties. You can see how I've done it here:
pom.xml: https://github.com/Telesphoreo/TotalFreedomMod/blob/1.13/pom.xml
TotalFreedomMod.java: https://github.com/Telesphoreo/TotalFreedomMod/blob/1.13/src/main/java/me/totalfreedom/totalfreedommod/TotalFreedomMod.java

Wild1145 commented:
Thanks @Telesphoreo - I'll take a look at implementing such an improvement as a separate issue for the 1.13 release.

Wild1145 committed:
POM Version updates & General fixes to resolve build

Swithed over to using TF-WorldEdit from Telesphoreo while we get the official repo up to date which also allowed some technical debt to be cleaned up. Fixed other build issues in the process.

Wild1145 committed:
Removed extra whitespace from POM
